src/fights.ts

```diff
@@ -1356,6 +1383,36 @@ const freeFightSources = [
       macroAllowsFamiliarActions: false,
     }
   ),
+  new FreeFight(
+    () => {
+      if (!have($item`closed-circuit pay phone`)) return false;
+      else if (have($effect`Shadow Affinity`)) return true;
+      else if (get("_shadowAffinityToday", false)) return false;
+
+      if (rufusTarget() === null) return true;
+      else if (get("rufusQuestType", "") === "items") {
+        return false; // We deemed it unprofitable to complete the quest in potionSetup
+      } else if (get("encountersUntilSRChoice", 0) === 0) {
+        // Target is either an artifact or a boss
+        return true; // Get the artifact or kill the boss immediately for free
+      }
+      return false; // We have to spend turns to get the artifact or kill the boss
```

Comment:
imo this section will read better if we strip the `else`s; early returns make them unnecessary